lib/Graph/TensorLayout.cpp

```diff
@@ -0,0 +1,277 @@
+static TensorDimensionDescription dimsNHWC[] = {
+    {.alignment = 1, .order = 0, .name = 'N'},
+    {.alignment = 1, .order = 1, .name = 'H'},
+    {.alignment = 1, .order = 2, .name = 'W'},
+    {.alignment = 1, .order = 3, .name = 'C'},
+};
+static TensorDimensionDescription dimsNCHW[] = {
+    {.alignment = 1, .order = 0, .name = 'N'},
+    {.alignment = 1, .order = 1, .name = 'C'},
+    {.alignment = 1, .order = 2, .name = 'H'},
+    {.alignment = 1, .order = 3, .name = 'W'},
+};
+static TensorDimensionDescription dimsHWNC[] = {
+    {.alignment = 1, .order = 0, .name = 'H'},
+    {.alignment = 1, .order = 1, .name = 'W'},
+    {.alignment = 1, .order = 2, .name = 'N'},
+    {.alignment = 1, .order = 3, .name = 'C'},
+};
+static TensorDimensionDescription dims1D[] = {
+    {.alignment = 1, .order = 0, .name = 'N'},
+};
+static TensorDimensionDescription dims2D[] = {
+    {.alignment = 1, .order = 0, .name = '*'},
+    {.alignment = 1, .order = 1, .name = '*'},
+};
+static TensorDimensionDescription dims3D[] = {
+    {.alignment = 1, .order = 0, .name = '*'},
+    {.alignment = 1, .order = 1, .name = '*'},
+    {.alignment = 1, .order = 2, .name = '*'},
+};
+static TensorDimensionDescription dims4D[] = {
+    {.alignment = 1, .order = 0, .name = '*'},
+    {.alignment = 1, .order = 1, .name = '*'},
+    {.alignment = 1, .order = 2, .name = '*'},
+    {.alignment = 1, .order = 3, .name = '*'},
+};
+static TensorDimensionDescription dims5D[] = {
+    {.alignment = 1, .order = 0, .name = '*'},
+    {.alignment = 1, .order = 1, .name = '*'},
+    {.alignment = 1, .order = 2, .name = '*'},
+    {.alignment = 1, .order = 3, .name = '*'},
+    {.alignment = 1, .order = 4, .name = '*'},
+};
+static TensorDimensionDescription dims6D[] = {
+    {.alignment = 1, .order = 0, .name = '*'},
+    {.alignment = 1, .order = 1, .name = '*'},
+    {.alignment = 1, .order = 2, .name = '*'},
+    {.alignment = 1, .order = 3, .name = '*'},
+    {.alignment = 1, .order = 4, .name = '*'},
+    {.alignment = 1, .order = 5, .name = '*'},
+};
+
+static TensorLayoutDescription layoutNHWC(dimsNHWC);
+static TensorLayoutDescription layoutNCHW(dimsNCHW);
+static TensorLayoutDescription layoutHWNC(dimsHWNC);
+static TensorLayoutDescription layout0D({});
+static TensorLayoutDescription layout1D(dims1D);
+static TensorLayoutDescription layout2D(dims2D);
+static TensorLayoutDescription layout3D(dims3D);
+static TensorLayoutDescription layout4D(dims4D);
+static TensorLayoutDescription layout5D(dims5D);
+static TensorLayoutDescription layout6D(dims6D);
+
+/// Glow layouts for any specific number of dimensions.
+static std::array<TensorLayoutDescription, max_tensor_dimensions + 1>
+    layoutsForDims = {
+        layout0D, layout1D, layout2D, layout3D, layout4D, layout5D, layout6D,
+};
+
+static std::unordered_map<std::string, TensorLayoutDescription *>
+    layoutNameToLayoutDescription = {{
+        {"NCHW", &layoutNCHW},
+        {GLOW_DEFAULT_4D_LAYOUT, &layoutNHWC},
```

Comment:
`GLOW_DEFAULT_4D_LAYOUT` should not be used here, because if we ever change the default, this line becomes incorrect. Please use a string here.